examples/dominantspeaker/src/index.js

```diff
@@ -0,0 +1,182 @@
+const roomNameText = document.querySelector('#roomName');
+const mediaContainer = document.getElementById('remote-media');
+const userControls = document.getElementById('user-controls');
+let roomName = null;
+const SAMPLE_USER_COUNT = 4;
+
+/**
+ * creates a button and adds to given container.
+ */
+function createButton(text, container) {
+  const btn = document.createElement('button');
+  btn.innerHTML = text; // 'Disconnect';
+  btn.classList.add('btn', 'btn-outline-primary', 'btn-sm');
+  container.appendChild(btn);
+  return btn;
+}
+
+/**
+ *
+ * creates controls for user to mute/unmute and disconnect
+ * from the room.
+ */
+async function createUserControls(userIdentity) {
+  const creds = await getRoomCredentials(userIdentity);
+  let room = null;
+
+  const currentUserControls = document.createElement('div');
+  currentUserControls.classList.add('usercontrol');
+
+  const title = document.createElement('h6');
+  title.appendChild(document.createTextNode(creds.identity));
+  currentUserControls.appendChild(title);
+
+  // connect button
+  const connectDisconnect = createButton('Connect', currentUserControls);
+  connectDisconnect.onclick = async function(event) {
+    connectDisconnect.disabled = true;
+    const connected = room !== null;
+    if (connected) {
+      room.disconnect();
+      room = null;
+      muteBtn.innerHTML = 'Mute';
+    } else {
+      room = await connectToRoom(creds);
+    }
+    connectDisconnect.innerHTML = connected ? 'Connect' : 'Disconnect';
+    muteBtn.style.display = connected ? 'none' : 'inline';
+    connectDisconnect.disabled = false;
+  }
+
+  // mute button.
+  const muteBtn = createButton('Mute', currentUserControls);
+  muteBtn.onclick = function() {
+    const mute = muteBtn.innerHTML == 'Mute';
+    const localUser = room.localParticipant;
+    getTracks(localUser).forEach(function(track) {
+      if (track.kind === 'audio') {
+        if (mute) {
+          track.disable();
+        } else {
+          track.enable();
+        }
+      }
+    });
+    muteBtn.innerHTML = mute ? 'Unmute' : 'Mute';
+  }
+  muteBtn.style.display = 'none';
+  userControls.appendChild(currentUserControls);
+}
+
+/**
+ * Connect the Participant with media to the Room.
+ */
+async function connectToRoom(creds) {
+  const room = await Video.connect( creds.token, {
+    name: roomName
+  });
+
+  return room;
+}
+
+/**
+ * Get the Tracks of the given Participant.
+ */
+function getTracks(participant) {
+  return Array.from(participant.tracks.values()).filter(function(publication) {
+    return publication.track;
+  }).map(function(publication) {
+    return publication.track;
+  });
+}
+
+/**
+ * add/removes css attribute per dominant speaker change.
+ * @param {Participant} speaker - Participant
+ * @param {boolean} add - boolean true when new speaker is detected. false for old speaker
+ * @returns {void}
+ */
+function updateDominantSpeaker(speaker, add) {
```

Comment:
@makarandp0 I think you can use one call to `updateDominantSpeaker()` to change the active speaker if you remove the "dominant_speaker" class from the existing Participant div before adding it to the new one:
```js
function updateDominantSpeaker(participant) {
  const dominantSpeakerDiv = document.querySelector('div.dominant_speaker');
  if (dominantSpeakerDiv) {
    dominantSpeakerDiv.classList.remove('dominant_speaker');
  }
  if (participant) {
    const newDominantSpeakerDiv = document.getElementById(participant.sid);
    if (newDominantSpeakerDiv) {
      newDominantSpeakerDiv.classList.add('dominant_speaker');
    }
  }
}
```

Reply:
thanks for the tip. that does look better.